Zend/zend_objects_API.c

```diff
@@ -120,6 +123,9 @@ ZEND_API void ZEND_FASTCALL zend_objects_store_free_object_storage(zend_objects_
 					GC_ADD_FLAGS(obj, IS_OBJ_FREE_CALLED);
 					GC_ADDREF(obj);
 					obj->handlers->free_obj(obj);
+					if (!GC_DELREF(obj)) {
+						zend_objects_store_del(obj);
```

Comment:
This (and the above call) can maybe be optimized with a specialised version, because e.g. overwriting the valid flag in the handle table is useless, and the check for IS_OBJ_FREE_CALLED also isn't necessary anymore. But I'm not sure that's worth it, or if this whole patch is worth it in the first place...